ArrayIndexOutOfBoundsException when using com.sun.xml.bind.v2.runtime.output.C14nXmlOutput

Hi,

The `com.sun.xml.bind.v2.runtime.output.C14nXmlOutput` class seems to have an out of bounds error:

```
Caused by: java.lang.ArrayIndexOutOfBoundsException: Index 8 out of bounds for length 8
	at com.sun.xml.bind.v2.runtime.output.C14nXmlOutput.attribute(C14nXmlOutput.java:126)
	at com.sun.xml.bind.v2.runtime.XMLSerializer.attribute(XMLSerializer.java:416)
	at com.sun.xml.bind.v2.runtime.property.AttributeProperty.serializeAttributes(AttributeProperty.java:73)
	at com.sun.xml.bind.v2.runtime.ClassBeanInfoImpl.serializeAttributes(ClassBeanInfoImpl.java:352)
	at com.sun.xml.bind.v2.runtime.XMLSerializer.childAsXsiType(XMLSerializer.java:659)
	at com.sun.xml.bind.v2.runtime.property.SingleElementNodeProperty.serializeBody(SingleElementNodeProperty.java:128)
	at com.sun.xml.bind.v2.runtime.ClassBeanInfoImpl.serializeBody(ClassBeanInfoImpl.java:329)
	at com.sun.xml.bind.v2.runtime.XMLSerializer.childAsXsiType(XMLSerializer.java:666)
	at com.sun.xml.bind.v2.runtime.property.ArrayElementNodeProperty.serializeItem(ArrayElementNodeProperty.java:39)
	at com.sun.xml.bind.v2.runtime.property.ArrayElementProperty.serializeListBody(ArrayElementProperty.java:142)
	at com.sun.xml.bind.v2.runtime.property.ArrayERProperty.serializeBody(ArrayERProperty.java:129)
	at com.sun.xml.bind.v2.runtime.ClassBeanInfoImpl.serializeBody(ClassBeanInfoImpl.java:329)
	at com.sun.xml.bind.v2.runtime.XMLSerializer.childAsSoleContent(XMLSerializer.java:563)
	at com.sun.xml.bind.v2.runtime.ClassBeanInfoImpl.serializeRoot(ClassBeanInfoImpl.java:310)
	at com.sun.xml.bind.v2.runtime.XMLSerializer.childAsRoot(XMLSerializer.java:464)
	at com.sun.xml.bind.v2.runtime.MarshallerImpl.write(MarshallerImpl.java:298)
	at com.sun.xml.bind.v2.runtime.MarshallerImpl.marshal(MarshallerImpl.java:226)
	at javax.xml.bind.helpers.AbstractMarshallerImpl.marshal(AbstractMarshallerImpl.java:110)
```

This class is used whenever the `com.sun.xml.bind.c14n property` is set to `true` and the `jaxb.formatted.output` property is set to `false`.

The offending code seems to be the highlighted bit here in `com.sun.xml.bind.v2.runtime.output.C14nXmlOutput#attribute(com.sun.xml.bind.v2.runtime.Name, java.lang.String)`:

```
    @Override
    public void attribute(Name name, String value) throws IOException {
        if(staticAttributes.length==len) {
            // reallocate
            int newLen = len*2;
            StaticAttribute[] newbuf = new StaticAttribute[newLen];
            System.arraycopy(staticAttributes,0,newbuf,0,len);
            for(int i=len;i<newLen;i++)
                staticAttributes[i] = new StaticAttribute(); <--- This should read newBuf[i] = new StaticAttribute();
            staticAttributes = newbuf;
        }

        staticAttributes[len++].set(name,value);
    }
```

Thoughts? Is there a workaround for getting C14N formatted output without also setting `jaxb.formatted.output` to `true`  to avoid this problem?

--- a/jaxb-ri/docs/release-documentation/src/docbook/jaxb-changelog.xml
+++ b/jaxb-ri/docs/release-documentation/src/docbook/jaxb-changelog.xml
@@ -41,6 +41,9 @@
                     <listitem><para>
                         <link xlink:href="https://github.com/eclipse-ee4j/jaxb-ri/issues/1450">#1450</link>: XJC -relaxng still supported
                     </para></listitem>
+                    <listitem><para>
+                        <link xlink:href="https://github.com/eclipse-ee4j/jaxb-ri/issues/1700">#1700</link>: ArrayIndexOutOfBoundsException when using com.sun.xml.bind.v2.runtime.output.C14nXmlOutput
+                    </para></listitem>
                 </itemizedlist>
             </para></listitem>
         </itemizedlist>

--- a/jaxb-ri/runtime/impl/src/main/java/org/glassfish/jaxb/runtime/v2/runtime/output/C14nXmlOutput.java
+++ b/jaxb-ri/runtime/impl/src/main/java/org/glassfish/jaxb/runtime/v2/runtime/output/C14nXmlOutput.java
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 1997, 2021 Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 1997, 2023 Oracle and/or its affiliates. All rights reserved.
  *
  * This program and the accompanying materials are made available under the
  * terms of the Eclipse Distribution License v. 1.0, which is available at
@@ -124,8 +124,9 @@ public class C14nXmlOutput extends UTF8XmlOutput {
             int newLen = len*2;
             StaticAttribute[] newbuf = new StaticAttribute[newLen];
             System.arraycopy(staticAttributes,0,newbuf,0,len);
-            for(int i=len;i<newLen;i++)
-                staticAttributes[i] = new StaticAttribute();
+            for(int i=len;i<newLen;i++) {
+                newbuf[i] = new StaticAttribute();
+            }
             staticAttributes = newbuf;
         }
 

